packages/plugins/src/plugins/balancer/getPoolTokensJob.ts

```diff
@@ -25,7 +29,29 @@ export function getPoolTokensJob(
       }, {} as Record<string, (typeof balancerTokens)[number]>)
     );
 
-    const priceSources = uniqueTokens
+    const poolPriceSources: TokenPriceSource[] = poolTokensRes.map((pool) => ({
+      address: formatTokenAddress(pool.address, networkId),
+      decimals: pool.decimals,
+      id: platformId,
+      networkId,
+      platformId,
+      price: BigNumber(pool.dynamicData.totalLiquidity)
+        .dividedBy(pool.dynamicData.totalShares)
+        .toNumber(),
+      underlyings: pool.poolTokens.map((token) => ({
+        networkId,
+        address: formatTokenAddress(token.address, networkId),
+        decimals: token.decimals,
+        price: BigNumber(token.balanceUSD).dividedBy(token.balance).toNumber(),
+        amountPerLp: BigNumber(token.balance)
+          .dividedBy(pool.dynamicData.totalShares)
+          .toNumber(),
+      })),
+      timestamp: Date.now(),
```

Comment:
Shouldn't the value be defined once outside the `.map()` and re-used ?
```suggestion
      timestamp,
```